### DISCUSSION TITLE
Emoji control Marker

### DISCUSSION THREAD

**Discussion by Ahmedgis1** [geoman-io/leaflet-geoman#1054] [Questions & Help] [answered]
Hello,

First of all Many Thanks for the great effort!

I want to have Emojis control Marker (so the user can add emojis to the maps as marker or point) and then use the coordinates of these points.
i am trying to have more than one marker Control. I used the function map.pm.addControls and enabled the marker and disable all other controls and repeated the function and put icons, but the Marker control on the map still one.

Is there any suggestions?

**Comment by Falke-Design** [comment, marked as answer]
You need to copy the marker multiple times and then add each one a custom icon with `markerStyle`.

css:
```
.icon1-button {
  background-image: url('https://leafletjs.com/examples/custom-icons/leaf-red.png');
}
```

Js:
```
var icon1 = L.icon({
iconUrl: 'https://leafletjs.com/examples/custom-icons/leaf-red.png'
})

map.pm.Toolbar.copyDrawControl('Marker', {
  name: 'MarkerCopy',
  block: 'custom',
  title: 'Display text on hover button',
  className: 'icon1-button'
});
map.pm.Draw.MarkerCopy.setOptions({markerStyle: {icon: icon1}});
```
https://jsfiddle.net/dkjrLxsq/

**Reply by Ahmedgis1** [reply]
Thanks @Falke-Design for your fast answer.

I have tryed a lot to use this: map.pm.CustomMarker.setOptions({markerStyle: {icon: emojiIcon }})
but everytime it gives me this error : "<a class='gotoLine' href='#94:15'>94:15</a> Uncaught TypeError: Cannot read properties of undefined (reading 'setOptions')"

I found in one of your answers on a question how you changed the icon of the Marker on the map, but also I want to know how t change the icon of the ControlMarker itself?

**Reply by Falke-Design** [reply]
Sorry, the correct way is `map.pm.Draw.CustomMarker.setOptions`. I updated the original answer with code and a sample

**Reply by Ahmedgis1** [reply]
Many Thanks. Your Answer was very helpful.....
